sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -980,19 +991,27 @@ private class BulkWriter(container: CosmosAsyncContainer,
                   numberOfIntervalsWithIdenticalActiveOperationSnapshots
                 )
 
-                if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get == 1L &&
+                if (numberOfIntervalsWithIdenticalActiveOperationSnapshots.get <= 10L &&
                   // pending retries only tracks the time to enqueue
                   // the retry - so this should never take longer than 1 minute
                   pendingRetriesSnapshot == 0L) {
+
+                  if (verboseLoggingAfterReEnqueueingRetriesEnabled.compareAndSet(false, true)) {
```

Comment:
`Scannable.from(bulkInputEmitter).scan(Scannable.Attr.BUFFERED)` may also be helpful here as it can tell us whether the item is still in the buffer

Reply:
ACK - fixed